packages/app-builder/src/routes/_builder+/d3.tsx

```diff
@@ -0,0 +1,251 @@
+  const { t } = useTranslation(['decisions']);
+
+  return (
+    <div className="flex flex-col gap-8">
+      <div className="relative size-full h-[500px]" ref={ref}>
+        {bounds.width > 0 ? (
+          <ChartInner
+            data={entries.nodes}
+            width={bounds.width}
+            height={bounds.height}
+          />
+        ) : null}
+      </div>
+      <div className="flex w-full flex-wrap items-center justify-center gap-6">
+        {orderredOutcomes.map((outcome) => (
+          <div key={outcome} className="inline-flex items-center gap-2">
+            <svg
+              className={clsx('size-4 shrink-0 rounded', outcomeColor[outcome])}
+            >
+              <rect
+                width="100%"
+                height="100%"
+                rx={4}
+                ry={4}
+                fill="currentColor"
+                className={outcomeColor[outcome]}
+              />
+            </svg>
+            <span className="text-s text-grey-100 font-medium">
+              {t(`decisions:outcome.${outcome}`)}
+            </span>
+          </div>
+        ))}
+      </div>
+    </div>
+  );
+}
+
+const orderredOutcomes: Outcome[] = [
+  'decline',
+  'block_and_review',
+  'review',
+  'approve',
+];
+
+const outcomeColor = {
+  approve: 'text-green-100',
+  decline: 'text-red-100',
+  review: 'text-yellow-100',
+  block_and_review: 'text-orange-100',
+  unknown: 'text-grey-100',
+};
+
+function ChartInner({
+  data,
+  width,
+  height,
+}: {
+  data: Data['nodes'];
+  width: number;
+  height: number;
+}) {
+  const margin = {
+    top: 37,
+    right: 0,
+    bottom: 0,
+    left: 0,
+  };
+  const gap = 4;
+
+  const foo = d3.index(
+    data,
+    (d) => d.version,
+    (d) => d.outcome,
+  );
+
+  // Determine the series that need to be stacked.
+  const series = d3
+    .stack<[string, d3.InternMap<Outcome, Data['nodes'][number]>], Outcome>()
+    .offset(d3.stackOffsetExpand)
+    .keys(orderredOutcomes)
+    .order(d3.stackOrderNone)
+    .value(([, D], key) => {
+      return D.get(key)?.count ?? 0;
+    })(foo);
+
+  // Prepare the scales for positional and color encodings.
+  const xScale = d3
+    .scaleBand()
+    .domain(d3.union(data.map((d) => d.version)))
+    .range([margin.left, width - margin.right])
+    .paddingInner(0.1);
+
+  const versionOptions = new Map(
+    foo.entries().map(([version, value]) => {
+      const outcomes = d3.sort(
+        value.keys(),
+        (outcome) => -orderredOutcomes.indexOf(outcome),
+      );
+      const outcomeOffset = new Map(
+        outcomes.map((outcome, i) => [outcome, i * gap]),
+      );
+      const numberOfOutcomes = Object.keys(outcomes).length;
+
+      return [
+        version,
+        {
+          outcomeOffset,
+          yScale: d3
+            .scaleLinear()
+            .domain([0, 1])
+            .range([
+              height - (numberOfOutcomes - 1) * gap - margin.bottom,
+              margin.top,
+            ]),
```

Comment:
This weird computation is required to handle the "gaps" between each part of the stack. This depends on each stack so we need to compute an `yScale` per stack.
![image](https://github.com/user-attachments/assets/89b692ba-1c1e-4e35-82d2-d48c5cb48f16)